app/models/king.rb

```diff
@@ -1,2 +1,30 @@
 class King < Piece
+  def valid_moves?(position, destination)
+    moves = []
+    moves << [x_position, y_position + 1] #Up one
+    moves << [x_position, y_position - 1] #Down one
+    moves << [x_position + 1, y_position] #Right one
+    moves << [x_position - 1, y_position] #Left one
+    moves << [x_position - 1, y_position + 1] #Diaganol up to left
+    moves << [x_position - 1, y_position - 1] #Diagonal down to left
+    moves << [x_position + 1, y_position + 1] #Diagonal up to right
+    moves << [x_position + 1, y_position - 1] #Diagonal down to right
+    moves.select { |move| position(move) }
```

Comment:
You've almost got it. For this last line, you want it to be,

"Does [destination_x, destination_y] exist in this list of possible valid moves?"

See if you can get it, let me know if you need help.